File: array_1/MiddleWay.java
import java.util.Arrays;

/**
 * Given 2 int arrays, a and b, each length 3, return a new arraylength 2 containing
 * their middle elements.
 * <ul>
 *  <li>middleWay([1, 2, 3], [4, 5, 6]) → [2, 5]
 *  <li>middleWay([7, 7, 7], [3, 8, 0]) → [7, 8]
 *  <li>middleWay([5, 2, 9], [1, 4, 5]) → [2, 4]
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p146449
 * @since 17.0.1
 * @version 0.0.1
 */
public class MiddleWay {
    public static void main(String[] args) {
        int[] array1 = {1, 2, 3};
        int[] array2 = {4, 5, 6};
        int[] array3 = {7, 7, 7};
        int[] array4 = {3, 8, 0};
        int[] array5 = {5, 2, 9};
        int[] array6 = {1, 4, 5};
        System.out.println("middleWay(" + Arrays.toString(array1) + ", " + Arrays.toString(array2) + ") -> " + Arrays.toString(middleWay(array1, array2)));
        System.out.println("middleWay(" + Arrays.toString(array3) + ", " + Arrays.toString(array4) + ") -> " + Arrays.toString(middleWay(array3, array4)));
        System.out.println("middleWay(" + Arrays.toString(array5) + ", " + Arrays.toString(array6) + ") -> " + Arrays.toString(middleWay(array5, array6)));
    }

    /**
     * Given 2 int arrays, a and b, each length 3, return a new arraylength 2 containing
     * their middle elements.
     * 
     * @param a integer array where the middle element is used
     * @param b integer array where the middle element is used
     * @return new integer array consisting of the middle arrays of a and b
     * @since 0.0.1
     */
    public static int[] middleWay(int[] a, int[] b) {
        int[] newArray = {a[a.length-2], b[b.length-2]};
        return newArray;
    }
}

File: list_1/Middle_way.py
'''
Given 2 int arrays, a and b, each length 3, return a new array length 2
containing their middle elements.
middle_way([1, 2, 3], [4, 5, 6]) → [2, 5]
middle_way([7, 7, 7], [3, 8, 0]) → [7, 8]
middle_way([5, 2, 9], [1, 4, 5]) → [2, 4]

author LZ-FSDev
see https://codingbat.com/prob/p171011
'''

def middle_way(a, b):
    array = [a[int(len(a)/2)], b[int(len(b)/2)]]
    return array

array1 = [1, 2, 3]
array2 = [4, 5, 6]
array3 = [7, 7, 7]
array4 = [3, 8, 0]
array5 = [5, 2, 9]
array6 = [1, 4, 5]

print("middle_way(" + str(array1) + ", " + str(array2) + ") -> " \
         + str(middle_way(array1, array2)))
print("middle_way(" + str(array3) + ", " + str(array4) + ") -> " \
         + str(middle_way(array3, array4)))
print("middle_way(" + str(array5) + ", " + str(array6) + ") -> " \
         + str(middle_way(array5, array6)))
